core/src/services/azblob/backend.rs

```diff
@@ -145,6 +148,49 @@ impl AzblobBuilder {
         self
     }
 
+    /// Set server_side_encryption_customer_key of this backend.
+    pub fn server_side_encryption_customer_key(
+        &mut self,
+        server_side_encryption_customer_key: &str,
+    ) -> &mut Self {
+        if !server_side_encryption_customer_key.is_empty() {
+            self.server_side_encryption_customer_key =
+                Some(server_side_encryption_customer_key.to_string());
+        }
+
+        self
+    }
+
+    /// Set server_side_encryption_customer_key_sha256 of this backend.
+    pub fn server_side_encryption_customer_key_sha256(
+        &mut self,
+        server_side_encryption_customer_key_sha256: &str,
+    ) -> &mut Self {
+        if !server_side_encryption_customer_key_sha256.is_empty() {
+            self.server_side_encryption_customer_key_sha256 =
+                Some(server_side_encryption_customer_key_sha256.to_string());
+        }
+
+        self
+    }
+
+    /// Set server_side_encryption_customer_algorithm of this backend.
+    pub fn server_side_encryption_customer_algorithm(
+        &mut self,
+        server_side_encryption_customer_algorithm: &str,
+    ) -> &mut Self {
+        if !server_side_encryption_customer_algorithm.is_empty() {
+            if server_side_encryption_customer_algorithm != "AES256" {
```

Comment:
We can check this in `build` which we can return an error instead.

Reply:
move encryption_algorithm value checking into `build`